wcfsetup/install/files/lib/system/event/listener/SessionAccessLogListener.class.php

```diff
@@ -51,22 +52,28 @@ public function execute($eventObj, $className, $eventName, array &$parameters) {
 				$sessionLogID = $sessionLog->sessionLogID;
 			}
 			
-			// format request uri
+			// Fetch request URI + request ID (if available).
 			$requestURI = UserUtil::getRequestURI();
-			// remove directories
-			$URIComponents = explode('/', $requestURI);
-			$requestURI = array_pop($URIComponents);
-			// remove session url
-			$requestURI = preg_replace('/(?:\?|&)s=[a-f0-9]{40}/', '', $requestURI);
+			if ($requestId = \wcf\getRequestId()) {
+				$requestURI .= ' ('.$requestId.')';
+			}
+			
+			// Get controller name + the AJAX action.
+			$className = get_class($eventObj);
+			if ($eventObj instanceof AJAXInvokeAction) {
+				if (isset($_REQUEST['className']) && isset($_REQUEST['actionName'])) {
+					$className .= ' ('.$_REQUEST['className'].':'.$_REQUEST['actionName'].')';
+				}
+			}
 			
 			// save access
 			ACPSessionAccessLogEditor::create([
 				'sessionLogID' => $sessionLogID,
 				'ipAddress' => UserUtil::getIpAddress(),
 				'time' => TIME_NOW,
-				'requestURI' => $requestURI,
+				'requestURI' => substr($requestURI, 0, 255),
```

Comment:
This will cut off the request ID, if the request URI is really long. I would move the `substr` call to line 57-59 and add an additional `else` case to trim the "regular" URI.